mesh/mesh.go

```diff
@@ -239,23 +232,66 @@ func (msh *Mesh) GetLayer(i types.LayerID) (*types.Layer, error) {
 	return l, nil
 }
 
+type internalLayer struct {
+	layerID   types.LayerID
```

Comment:
fields need to be exported for serialisation

Reply:
done

we didn't check the number of bytes marshaled/unmarshaled so InterfaceToBytes/BytesToInterface fails silently in this case. i think we should at least print an warning if not return an error of zerobyte and let the caller decide whether it's expected...